### PR TITLE
Improve usage by Java consumers

### LINKED ISSUE #312: Usage from Java requires red code
The equivalent code for
```kotlin
Paparazzi(
    theme = "AppTheme.ScreenshotTest",
    maxPercentDifference = 0.0,
    renderExtensions = listOf(CropViewRenderExtension()),
)
```
used in a Java project:
```java
new Paparazzi(
		EnvironmentKt.detectEnvironment(),
		DeviceConfig.Companion.getNEXUS_5(),
		"AppTheme.ScreenshotTest",
		true,
		0.0,
		Boolean.parseBoolean(System.getProperty("paparazzi.test.verify"))
				? new SnapshotVerifier(0.0, new File("src/test/snapshots"))
				: new HtmlReportWriter(),
		new HashSet<>(Collections.singletonList(new CropViewRenderExtension()))
);
```
gives an error in the IDE (fine during compilation).
image

### PR DESCRIPTION
**Pull request by jrodbx**
Closes #312

### CODE REVIEW

**Review by JakeWharton — APPROVED** [review]

File: paparazzi/src/main/java/app/cash/paparazzi/HtmlReportWriter.kt [hunk 1]
```diff
@@ -57,7 +57,7 @@ import javax.imageio.ImageIO
  * paparazzi.js
  * ```
  */
-internal class HtmlReportWriter(
+class HtmlReportWriter @JvmOverloads constructor(
```
Comment (Egorand) [inline paparazzi/src/main/java/app/cash/paparazzi/HtmlReportWriter.kt:60]: Does this class need to be in the public API?

**Review by Egorand — APPROVED** [review]

File: paparazzi/src/main/java/app/cash/paparazzi/HtmlReportWriter.kt (hunk 1 above)
Comment (jrodbx) [reply paparazzi/src/main/java/app/cash/paparazzi/HtmlReportWriter.kt:60]: Good question.  In the referenced issue, the snapshot handler is the second-to-last arg, so the equivalent java overload makes it required to set this when setting RenderExtensions.

I _could_ swap the parameter ordering, but wasn't sure if that was good form, i.e., does that mean all future parameter additions need to not be last?

Comment (Egorand) [reply paparazzi/src/main/java/app/cash/paparazzi/HtmlReportWriter.kt:60]: Interesting. For Java we'd probably just want a separate `Builder`..

Comment (jrodbx) [reply paparazzi/src/main/java/app/cash/paparazzi/HtmlReportWriter.kt:60]: Ah true, didn't even think of providing _that_ level of support.  Isn't everyone on Kotlin by now?  😛 

I'll merge this for now and follow-up with Builder.

Comment (Egorand) [reply paparazzi/src/main/java/app/cash/paparazzi/HtmlReportWriter.kt:60]: We can maybe just say "sorry no explicit Java support"? IDK

**Pull request merged**